dissect/target/plugins/os/windows/lnk.py

```diff
@@ -72,14 +73,14 @@ def lnk(self, path: Optional[str]) -> Iterator[LnkRecord]:
         """
 
         # we need to get the active codepage from the system to properly decode some values
-        codepage = self.target.codepage
+        codepage = self.target.codepage or "ascii"
 
-        for entry in self.lnk_entries(path):
+        for entry in self.lnk_entries(directory):
             lnk_file = Lnk(entry.open())
             lnk_net_name = lnk_device_name = None
 
             if lnk_file.link_header:
-                lnk_path = str(entry)
+                lnk_path = path.from_windows(entry)
```

Comment:
```suggestion
                lnk_path = entry
```
Entry is a `(Target)Path` object so there should be no need to do this. Speaking of, can you change the return type of `lnk_entries` to `Iterator[Path]` instead?